components/raftstore-v2/src/raft/peer.rs

```diff
@@ -210,6 +211,43 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         Ok(peer)
     }
 
+    pub fn maybe_gen_approximate_buckets<T>(&self, ctx: &StoreContext<EK, ER, T>) {
+        if ctx.coprocessor_host.cfg.enable_region_bucket && !self.region().get_peers().is_empty() {
+            if let Err(e) = ctx
+                .schedulers
+                .split_check
+                .schedule(SplitCheckTask::ApproximateBuckets(self.region().clone()))
+            {
+                error!(
+                    self.logger,
+                    "failed to schedule check approximate buckets";
+                    "err" => %e,
+                );
+            }
+        }
+    }
+
+    #[inline]
+    pub fn region_buckets_mut(&mut self) -> &mut BucketStat {
```

Comment:
Why directly modify it instead of `set_region_buckets`?

Reply:
It needs to  clear the stats after reporting bucket to pd.

Reply:
I know why you used it, but I don't know why you didn't use `set_region_buckets`. To be more specific, for a new contributor wanting to mutate this field, which one should he use?

Reply:
yes, I will remove it.